src/components/Register/index.jsx

```diff
@@ -0,0 +1,252 @@
+        visible={termsOpen}
+        onOk={handleOkTerms}
+        onCancel={handleCancelTerms}
+        footer={[
+          <Button key="back" onClick={handleCancelTerms}>
+            {t("register.close")}
+          </Button>,
+          <Button key="submit" type="primary" onClick={handleOkTerms}>
+            {t("register.iAgree")}
+          </Button>,
+        ]}
+      >{t("register.agreeOnTerms",{ returnObjects: true,})}
+      </Modal>
+      <div className="loginContainer">
+        <div className="loginContainer__loginTitle">{t("register.creat")}</div>
+        {props.error && (
+          <Alert
+            style={{ marginBottom: 10 }}
+            type="error"
+            showIcon
+            message={props.error}
+          ></Alert>
+        )}
+        <div className="loginContainer__loginDialog">
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.name")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors["fullname"] &&
+                "loginContainer__loginDialog__input__hasError"
+              }
+              value={registerInformation.fullname}
+              onChange={(e) => {
+                handleChange("fullname", e.target.value);
+              }}
+            />
+            {errors["fullname"] && (
+              <div className="loginContainer__loginDialog__errorContainer">
+                {errors["fullname"]}
+              </div>
+            )}
+          </div>
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.enterAge")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors["age"] && "loginContainer__loginDialog__input__hasError"
+              }
+              type="number"
+              value={registerInformation.age}
+              onChange={(e) => {
+                handleChange("age", e.target.value);
+              }}
+            />
+            {errors["age"] && (
+              <div className="loginContainer__loginDialog__errorContainer">
+                {errors["age"]}
+              </div>
+            )}
+          </div>
+          <div className="loginContainer__loginDialog__inputLabel">
+            {t("register.email")}
+          </div>
+          <div className="loginContainer__loginDialog__input">
+            <Input
+              className={
+                errors["email"] &&
+                "loginContainer__loginDialog__input__hasError"
+              }
+              value={registerInformation.email}
+              onChange={(e) => {
+                handleChange("email", e.target.value);
+              }}
+            />
+            {errors["email"] && (
+              <div className="loginContainer__loginDialog__errorContainer">
+                {errors["email"]}
```

Comment:
it seems to me that lines 157 through 161 can be moved to a new component that takes a key

<ErrorMessage message={errors["email"]}/>

and this component will return null if errors["email"] is not present.

then you can remove a lot of copy-pasted code.

Reply:
I will do it.

Reply:
PTAL